Model conan

lllsy12138 opened the pull request:
<!-- If you are submitting a dataset or a model for the model registry please use the corresponding checklists below otherwise feel free to remove them. -->

<!-- add additional description, question etc. related to the new dataset -->


### Code Quality
<!-- Please do not delete this -->
- [x] **Code Formatted**: Format the code using `make lint` to maintain consistent style.

### Documentation
<!-- Please do not delete this -->
- [ ] **Updated Documentation**: Add or update documentation to reflect the changes introduced in this PR.

### Testing
<!-- Please do not delete this -->
- [ ] **New Tests Added**: Write tests to cover new functionality. Validate with `make test-with-coverage`.
- [ ] **Tests Passed**: Run tests locally using `make test` or `make test-with-coverage` to ensure no existing functionality is broken.

### Adding a model checklist
<!--
When adding a model to the model registry
see also https://github.com/embeddings-benchmark/mteb/blob/main/docs/reproducible_workflow.md
-->

 - [x] I have filled out the ModelMeta object to the extent possible
 - [x] I have ensured that my model can be loaded using
   - [x] `mteb.get_model(model_name, revision)` and
   - [x] `mteb.get_model_meta(model_name, revision)`
 - [x] I have tested the implementation works on a representative set of tasks.

We have added the model implementation and other information. Thanks.

Samoed commented on mteb/models/conan_models.py:
```diff
@@ -0,0 +1,207 @@
+
+logger = logging.getLogger(__name__)
+
+import time
+import logging
+import numpy as np
+from typing import Any
+import string
+import random
+import hashlib
+import json
+import requests
+import os
+from threading import Lock
```
Can you merge imports with impoerts from previous section?

Samoed commented on mteb/models/conan_models.py:
```diff
@@ -0,0 +1,207 @@
+        self.lock = Lock() 
+
+    def wait(self):
+        with self.lock:
```
I don't think you need to lock, because everything is running synchronously.

Samoed commented on mteb/models/conan_models.py:
```diff
@@ -0,0 +1,207 @@
+                    timeout=self.timeout, 
+                    headers=headers
+                )
+                result = json.loads(rsp.text)
```
```suggestion
                result = rsp.json()
```

Samoed commented on mteb/models/conan_models.py:
```diff
@@ -0,0 +1,207 @@
+                
+                return result
+
+            except Exception as e:
+                retries += 1
+                if retries < self.max_retries:
+                    sleep_time = 10 * retries 
+                    logger.warning(f"Request failed (attempt {retries}/{self.max_retries}), "
+                                 f"sleeping for {sleep_time}s. Error: {str(e)}")
+                    time.sleep(sleep_time)
+                else:
+                    logger.error(f"Max retries reached. Last error: {str(e)}")
+                    raise
```
Maybe move this code to RateLimiter?

lllsy12138 commented:
I've addressed the feedback:
- Fixed the code style issues
- Refactored the retry logic


I apologize for not writing better code in the first place. Thank you for your thorough review and valuable feedback.

Please review when you have time. Thank you!

Samoed commented on mteb/models/conan_models.py:
```diff
@@ -0,0 +1,211 @@
+            Exception: When max retries are reached and still failing
+        """
+        # First, wait for rate limiting
+        self.wait()
```
Maybe move wait in retry loop?

Samoed reviewed (commented):
Looks good. Can your lint your code with `make lint`?

lllsy12138 commented:
Thanks. I've moved wait() into the retry loop and the code has been linted.
